### PR TITLE
Reset connection deadline before connection is released

### PR DESCRIPTION
**Pull request by jkaflik**
## Summary

It's not well well-thought-out fix attempt for https://github.com/ClickHouse/clickhouse-go/issues/1607.

Requires deep dive into edge cases.

## Checklist
Delete items not relevant to your PR:
- [ ] Unit and integration tests covering the common scenarios were added
- [ ] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

File: conn_query.go [hunk 1]
```diff
@@ -80,6 +79,8 @@ func (c *connect) query(ctx context.Context, release nativeTransportRelease, que
 		}
 		close(stream)
 		close(errors)
+
+		c.conn.SetDeadline(time.Time{})
```
Comment (GeorgeMac) [inline conn_query.go:83]: I think we might need to do this for `SetReadDeadline` too. As the SetDeadline is an override on the context. I am unsure what we're using exactly in our case.

p.s. thanks for jumping on this so quickly!

**Comment by SpencerTorres** [comment]
Thanks for the info + discussion in this thread!

@GeorgeMac's solution is what we need. I have merged a snippet from his solution in https://github.com/ClickHouse/clickhouse-go/pull/1612 in https://github.com/ClickHouse/clickhouse-go/pull/1617
